app/models/customer.rb

```diff
@@ -28,6 +28,7 @@ class Customer < ActiveRecord::Base
 
   # gem 'paranoia' doesn't run validation callbacks on restore
   before_restore :run_validations
+  before_destroy :remove_from_transactions
```

Comment:
@yunixon why not use `dependent: :nullify`? http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html